subsys/partition_manager/Kconfig

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menuconfig PM_EXTERNAL_FLASH
```

Comment:
All this information is duplicated from device tree, is there any way we can deduce from device tree that external flash is hooked up?

Reply:
If a devicetree node for the flash exists, it will have 'existence flags' defined:

https://docs.zephyrproject.org/latest/guides/dts/macros.html#dt-existence-macros

Reply:
But how can we deduce that is flash in some generic way? By parsing the DT output we need to have some generic way of knowing that an external flash is specified, it is not clear to me how to do this.

Reply:
There are a couple of ways to go about this depending on your requirements:

- an alias, if you need a general flash device for some application specific purpose
- a chosen node, if it's a system wide setting, like we do with `zephyr,console`

What do you need?

Reply:
Hm, maybe a chosen node is what we need. Then the overlay specifying the SPI interface hooking up to the external flash could also specify that this is the chosen node for the external flash that should be used for partition placed in the "external flash" region.